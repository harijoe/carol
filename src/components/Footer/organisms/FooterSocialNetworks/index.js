import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import Section from 'components/Section'
import { theme, breakpoint } from 'utils/style'
import FooterSocialNetworksLink from '../../molecules/FooterSocialNetworksLink'

const StyledSection = styled(Section)`
  padding-bottom: ${theme('spaces.m')};
  padding-top: ${theme('spaces.m')};

  ${breakpoint('m')`
    padding-bottom: ${theme('spaces.l')};
    padding-top: ${theme('spaces.l')};
  `}

  ${breakpoint('xl')`
    padding-bottom: ${theme('spaces.l')};
    padding-top: ${theme('spaces.l')};
  `}
`

const FooterSocialNetworks = ({ locale }) =>
  <StyledSection>
    <FooterSocialNetworksLink id="facebook" locale={locale} />
    <FooterSocialNetworksLink id="twitter" locale={locale} />
    <FooterSocialNetworksLink id="linkedin" locale={locale} />
    <FooterSocialNetworksLink id="instagram" locale={locale} />
    <FooterSocialNetworksLink id="google" locale={locale} />
    <FooterSocialNetworksLink id="youtube" locale={locale} />
    <FooterSocialNetworksLink id="pinterest" locale={locale} />
  </StyledSection>

FooterSocialNetworks.propTypes = {
  locale: PropTypes.string,
}

export default FooterSocialNetworks
