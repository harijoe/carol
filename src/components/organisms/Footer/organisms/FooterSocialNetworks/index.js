import React from 'react'
import styled from 'styled-components'
import { Grid, Col, Row } from 'react-styled-flexboxgrid'

import { Icon, Link } from 'components'
import { theme } from 'utils/style'

const Wrapper = styled.div`
  width: 100%;
  padding: ${theme('spaces.m')};
  background-color: ${theme('colors.white')};
`

const StyledIcon = styled(Icon)`
  height: ${theme('sapces.l')};
  margin-right: ${theme('spaces.l')};
`

const FooterSocialNetworks = () => (
  <Wrapper>
    <Grid>
      <Row>
        <Col>
          <Link><StyledIcon icon="social-facebook" /></Link>
          <Link><StyledIcon icon="social-twitter" /></Link>
          <Link><StyledIcon icon="social-google" /></Link>
          <Link><StyledIcon icon="social-linkedin" /></Link>
        </Col>
      </Row>
    </Grid>
  </Wrapper>
)

export default FooterSocialNetworks
