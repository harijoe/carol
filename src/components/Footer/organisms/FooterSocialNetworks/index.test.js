import React from 'react'
import { shallow } from 'enzyme'
import mockIntl from 'mocks/intlMock'
import FooterSocialNetworks from '.'

it('renders the FooterSocialNetworks component', () => {
  const wrapper = shallow(mockIntl(<FooterSocialNetworks />))

  expect(wrapper).toMatchSnapshot()
})
