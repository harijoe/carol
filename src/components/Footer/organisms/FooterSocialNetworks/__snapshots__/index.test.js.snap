// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FooterSocialNetworks component 1`] = `<FooterSocialNetworks />`;
